### PR TITLE
Update english doc of develop guide and plugin introduction

### LINKED ISSUE #329: Translate official website documents from Chinese to English
### What would you like to be added?

Translate official website documents from Chinese to English

### Why is this needed?

The English and Chinese documents on the official website are not synchronized.

### PR DESCRIPTION
**Pull request by lilai23**
What type of PR is this?

doc

What this PR does / why we need it?

Update english doc of develop guide and plugin introduction

Which issue(s) this PR fixes:

Fixes #329

### CODE REVIEW

File: docs/en/document/developer-guide/bytecode-enhancement.md [hunk 1]
```diff
@@ -207,17 +207,19 @@ For [MethodMatcher](https://github.com/sermant-io/Sermant/blob/develop/sermant-a
...
+- Enhancing native classes may lead to unpredictable errors. Since native classes are widely used during development, enhancing them could potentially impact other plugins or host functions that rely on these classes.
+- Enhancing certain native class methods used during the enhancement of Sermant may result in class circular dependency errors. For example, the Sermant class loader uses the `java.net.URL` class when loading classes. If the constructor of the `URL` class is enhanced in a plugin, the JVM may throw a `ClassCircularityError` after the host application mounts the Agent and starts.
 
 To sum up, The ability to enhance *Java* native classes is provided in [**Sermant** core function module](https://github.com/sermant-io/Sermant/tree/develop/sermant-agentcore/sermant-agentcore-core), but it is not recommended to enhance them without restriction, if there are multiple enhancement points to choose from, the common class is preferred.
 
+> In the [Sermant Framework FAQ](../faq/framework.md), we have listed the known [native classes that Sermant does not support for enhancement](../faq/framework.md#sermant不支持对哪些原生类的增强).
+
```
Comment (daizhenyu) [inline docs/en/document/developer-guide/bytecode-enhancement.md:218]: Referenced Chinese Documents

Comment (lilai23) [reply docs/en/document/developer-guide/bytecode-enhancement.md:218]: modified

**Review by daizhenyu — APPROVED** [review]
